uwblueprint/starter-code-v2
# Add backend TS csv generation utility

## Notion ticket link

[TS Backend CSV generation utility](https://github.com/uwblueprint/starter-code-v2/issues/65)



## Implementation description
* Wrapper on the json2csv function similar to the frontend
* A response util function that either looks for an `application/json` or `text/csv` content-type to determine what type of response to send back


## Steps to test
REST (using Postman)
1. Set the content-type in the headers to `text/csv` and the request type to GET
2. Test `http://localhost:5000/entities` and `http://localhost:5000/users` to ensure you get back number of items in CSV format

Graphql (using Altair)
1. Set the request type to GET and put in this URL `http://localhost:5000/graphql`
2. Run the following queries and ensure you get the same CSV string as the REST responses

```
query {
   usersCSV
} // gets all users

query {
   entitiesCSV
} // should return all entities
```



## What should reviewers focus on?
* Might have missed this but do we need options similar to the frontend PR?


## Checklist
